src/vmm/src/device_manager/mmio.rs

```diff
@@ -151,32 +112,27 @@ impl MMIODeviceManager {
         MMIODeviceManager {
             mmio_base,
             next_avail_mmio: mmio_base,
-            irqs: IrqManager::new(irq_interval.0, irq_interval.1),
+            legacy_irq_allocator: IdAllocator::new(irq_interval.0, irq_interval.1).unwrap(),
             bus: devices::Bus::new(),
             id_to_dev_info: HashMap::new(),
         }
     }
 
     /// Allocates resources for a new device to be added.
     fn allocate_new_slot(&mut self, irq_count: u32) -> Result<MMIODeviceInfo> {
-        let irqs = self.irqs.get(irq_count)?;
+        let mut irqs: Vec<u32> = Vec::<u32>::new();
+        for _ in 0..irq_count {
+            let res = self.legacy_irq_allocator.allocate_id();
+            irqs.push(res.map_err(Error::AllocatorError)?);
+        }
```

Comment:
```suggestion
        let irqs: Result<Vec<u32>> = (0..irq_count)
            .map(|_| {
                self.legacy_irq_allocator
                    .allocate_id()
                    .map_err(Error::AllocatorError)
            })
            .collect();
```

And then, in MMIODeviceInfo initializer:

```
        let slot = MMIODeviceInfo {
            addr: self
                .address_allocator
                .allocate(MMIO_LEN, MMIO_LEN, AllocPolicy::FirstMatch)
                .map_err(Error::AllocatorError)?
                .start(),
            len: MMIO_LEN,
            irqs: irqs?,
        };
```